sktime/performance_metrics/forecasting/_functions.py

```diff
@@ -2625,23 +2627,50 @@ def relative_loss(
     multioutput=[0.3, 0.7])  # doctest: +SKIP
     np.float64(0.927272727272727)
     """  # noqa: E501
-    y_pred_benchmark = _get_kwarg(
-        "y_pred_benchmark", metric_name="relative_loss", **kwargs
-    )
+    # y_pred_benchmark = _get_kwarg(
+    #     "y_pred_benchmark", metric_name="relative_loss", **kwargs
+    # )
     _, y_true, y_pred, multioutput = _check_reg_targets(y_true, y_pred, multioutput)
+    _, _, y_pred_benchmark, _ = _check_reg_targets(
+        y_true, y_pred_benchmark, multioutput
+    )
 
     if horizon_weight is not None:
         check_consistent_length(y_true, horizon_weight)
 
-    loss_preds = relative_loss_function(
-        y_true, y_pred, horizon_weight=horizon_weight, multioutput=multioutput
-    )
-    loss_benchmark = relative_loss_function(
-        y_true,
-        y_pred_benchmark,
-        horizon_weight=horizon_weight,
-        multioutput=multioutput,
-    )
+    # dispatch on metric object vs bare function
+
+    rel_fn = loss_function if loss_function is not None else relative_loss_function
+
+    if isinstance(rel_fn, BaseForecastingErrorMetric):
```

Comment:
I would factor this out into a coercion function - we can also use `_DynamicForecastingMetric` to handle functions.